architecture/additional_concepts/authentication.adoc

```diff
@@ -345,6 +345,205 @@ desired behavior:
++
+Add the following sample code:
++
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: proxy
+spec:
+  to:
+    name: proxy
+  tls:
+    termination: Reencrypt
+apiVersion: v1
+kind: Service
+metadata:
+  name: proxy
+  annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: proxy-tls
+spec:
+  ports:
+  - name: proxy
+    port: 443
+    targetPort: 8443
+  selector:
+    app: proxy
+----
+
+.. Create a YAML for a deployment config to launch a proxy as a sidecar:
++
+----
+vi proxysidecar.yaml
+----
++
+Add the following sample code:
++
+----
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: proxy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: proxy
+  template:
+    metadata:
+      labels:
+        app: proxy
+    spec:
+      serviceAccountName: proxy
+      containers:
+      - name: oauth-proxy
+        image: openshift/oauth-proxy:v1.0.0
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 8443
+          name: public
+        args:
+        - --https-address=:8443
+        - --provider=openshift
+        - --openshift-service-account=proxy
+        - --upstream=http://localhost:8080
+        - --tls-cert=/etc/tls/private/tls.crt
+        - --tls-key=/etc/tls/private/tls.key
+        - --cookie-secret=SECRET
+        volumeMounts:
+        - mountPath: /etc/tls/private
+          name: proxy-tls
+
+      - name: app
+        image: openshift/hello-openshift:latest
+      volumes:
+      - name: proxy-tls
+        secret:
+          secretName: proxy-tls
+----
++
+.. Create the objects
++
+----
+oc create -f serviceaccount.yaml
+oc create -f route.yaml
+oc create -f proxysidecar.yaml
+----
+
+. Run `oc edit sa/proxy` to edit the service account and change the `serviceaccounts.openshift.io/oauth-redirectreference` annotation to point to a Route that does not exist.
++
+----
+apiVersion: v1
+imagePullSecrets:
+- name: proxy-dockercfg-08d5n
+kind: ServiceAccount
+metadata:
+  annotations:
+    serviceaccounts.openshift.io/oauth-redirectreference.primary: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"notexist"}}'
+...
+----
+
+. Browse to the service and click through the OAuth login flow to encounter the server error:
++
+----    
+The authorization server encountered an unexpected condition that prevented it from fulfilling the request.
+----    
+
+. Run `oc get events` to view the ServiceAccount event
++
+----
+23m        23m         1         proxy                    ServiceAccount                                  Warning   NoSAOAuthRedirectURIs   service-account-oauth-client-getter   [routes.route.openshift.io "notexist" not found, system:serviceaccount:myproject:proxy has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>]              
+----
```

Comment:
There should probably be a step after this that states that fixing the annotation would then lead to a functional OAuth flow.